addons/pos_self_order/static/src/app/models/order.js

```diff
@@ -55,8 +55,8 @@ export class Order extends Reactive {
         if (this.pos_reference) {
             const reference = this.pos_reference;
             const arrRef = reference.split(" ")[1].split("-");
-            const sessionID = parseInt(arrRef[0]).toString();
-            const sequence = parseInt(arrRef[2]).toString();
+            const sessionID = arrRef[0][4];
+            const sequence = arrRef[2].substr(2, 2);
             const trackingNumber = sessionID + sequence;
             return trackingNumber;
         }
```

Comment:
This will not works if the number is bigger than 1 char.

Try maybe something like that:

```
const reference = this.pos_reference;
const arrRef = reference.split(" ")[1].split("-");
const trackingNumber = `${parseInt(arrRef[0])}${parseInt(arrRef[2])}`;
return trackingNumber;
```